app/boards/shields/kyria/keymaps/default/keymap.overlay

```diff
@@ -0,0 +1,50 @@
+#include <dt-bindings/zmk/keys.h>
+#include <dt-bindings/zmk/keymap-transform.h>
+#include <keymap.h>
+
+/ {
+	chosen {
+		zmk,keymap = &keymap0;
+	};
+	
+	keymap0: keymap {
+		compatible = "zmk,keymap";
+		label ="Default Kyria Keymap";
+		layers = <&default>;
+		transform = <&default_transform>;
+	};
+
+	default_transform: keymap_transform_0 {
+		compatible = "zmk,keymap-transform";
+		map = <
+			IDENTITY_ROW(0, 6)       OFFSET_ROW(6, 0, 6)
+			IDENTITY_ROW(1, 6)       OFFSET_ROW(6, 1, 6)
+			IDENTITY_ROW(2, 8)       OFFSET_ROW(8, 2, 8)
+			OFFSET_ROW(3, 3, 5)      OFFSET_ROW(8, 3, 5)
+		>;
+	};
+
+	five_column_transform: keymap_transform_1 {
```

Comment:
This will *nicely* let us define transforms for keyboards that have breakaway PCB sections, so you can have your user config select the different transform, and now you can use the keymap that doesn't list that extra keys that aren't actually there for your physical keyboard.

Reply:
I like the final result of the keys=<>. but the map with offset row helpers seems a bit un-intuitive to me compared to a defined function that takes the keys in as parameters (and can be styled however with white space), and then the function returns the proper matrix with the keys in the right place. I imagine you might be constrained with these files though.

Reply:
Ok, so after discussion on Discord, this now looks like this:

![Screenshot from 2020-06-16 21-29-22](https://user-images.githubusercontent.com/202695/84845373-811cad80-b01a-11ea-96da-58f6001995ab.png)

using just one macro `RC` to map a logical location to a matrix row + column.